src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -166,6 +169,19 @@ internal override bool IsFilterSet()
             return (attributeFilterSet || base.IsFilterSet());
         }
 
+        /// <inheritdoc />
+        protected override bool TrySetCodeProperties(object item, string path)
+        {
+            if (PSObject.Base(item) is FileSystemInfo fileSystemInfo)
```

Comment:
Should we enhance and use `public enum ProviderCapabilities`

Reply:
Seems like a good idea!